frontend/src/components/RegisterUser/reducer.tsx

```diff
@@ -0,0 +1,61 @@
+import { AnyAction } from 'redux';
+
+export interface IRegisteredUserState {
+  email?: string;
+  fullname?: string;
+  id?: number;
+  loading: boolean;
+  masterStatus?: string;
+  registered: boolean;
+  username?: string;
+}
+
+const initialState = {
+  email: undefined,
+  fullname: undefined,
+  id: undefined,
+  loading: false,
+  masterStatus: undefined,
+  registered: false,
+  username: undefined,
+};
+
+export const registeredUserReducer = (
+    state: IRegisteredUserState = initialState, action: AnyAction,
+  ): IRegisteredUserState => {
+  const { type, payload } = action;
+  switch (type) {
+    case 'SET_USER_DATA': {
+      return {
+        ...state,
+        email: payload.email,
```

Comment:
should maybe be written to
{
...state,
...payload,
loading: false,
registered: true
}